### PR TITLE
BUG: Avoid redshift slider warning when setting y_limits

### LINKED ISSUE #2029: [BUG] Suppress redshift slider warning when setting cubeviz.specviz y limits
### Jdaviz component

Cubeviz

### Description

When using the API to set the cubeviz.specviz ylimits , there is an unexpected warning regarding the "redshift slider", which I am not even using

### How to Reproduce

1. Load a data cube in Cubeviz
2. Set the y-limits in the spectrum viewer using the API:
3. cubeviz.specviz.y_limits(0.0, 2.0E5)
4. A warning is given regarding the redshift slider:

/Users/pogle/miniconda3/envs/science/lib/python3.10/site-packages/jdaviz/configs/specviz/helper.py:88: UserWarning: Applying the value from the redshift slider to the output spectra. To avoid seeing this warning, explicitly set the apply_slider_redshift keyword option to True or False.
  warnings.warn("Applying the value from the redshift "

### Expected behavior

A warning should not be thrown when setting ylimits, at least not when the redshift slider has not even been used.

### Browser

Firefox

### Jupyter

jupyter --version
Selected Jupyter core packages...
IPython          : 8.5.0
ipykernel        : 6.16.0
ipywidgets       : 7.7.2
jupyter_client   : 7.3.5
jupyter_core     : 4.11.1
jupyter_server   : 1.19.1
jupyterlab       : 3.4.7
nbclient         : 0.5.13
nbconvert        : 6.5.4
nbformat         : 5.6.1
notebook         : 6.4.12
qtconsole        : 5.3.2
traitlets        : 5.4.0

### Software versions

import platform; print(platform.platform())
import sys; print("Python", sys.version)
import numpy; print("Numpy", numpy.__version__)
import astropy; print("astropy", astropy.__version__)
import matplotlib; print("matplotlib", matplotlib.__version__)
import scipy; print("scipy", scipy.__version__)
import skimage; print("scikit-image", skimage.__version__)
import asdf; print("asdf", asdf.__version__)
import gwcs; print("gwcs", gwcs.__version__)
import regions; print("regions", regions.__version__)
import specutils; print("specutils", specutils.__version__)
import specreduce; print("specreduce", specreduce.__version__)
import photutils; print("photutils", photutils.__version__)
import astroquery; print("astroquery", astroquery.__version__)
import yaml; print("pyyaml", yaml.__version__)
import asteval; print("asteval", asteval.__version__)
import idna; print("idna", idna.__version__)
import traitlets; print("traitlets", traitlets.__version__)
import bqplot; print("bqplot", bqplot.__version__)
import bqplot_image_gl; print("bqplot-image-gl", bqplot_image_gl.__version__)
import glue; print("glue-core", glue.__version__)
import glue_jupyter; print("glue-jupyter", glue_jupyter.__version__)
import glue_astronomy; print("glue-astronomy", glue_astronomy.__version__)
import echo; print("echo", echo.__version__)
import ipyvue; print("ipyvue", ipyvue.__version__)
import ipyvuetify; print("ipyvuetify", ipyvuetify.__version__)
import ipysplitpanes; print("ipysplitpanes", ipysplitpanes.__version__)
import ipygoldenlayout; print("ipygoldenlayout", ipygoldenlayout.__version__)
import jinja2; print("Jinja2", jinja2.__version__)
import voila; print("voila", voila.__version__)
import vispy; print("vispy", vispy.__version__)
import sidecar; print("sidecar", sidecar.__version__)
import jdaviz; print("Jdaviz", jdaviz.__version__)

macOS-11.7-x86_64-i386-64bit
Python 3.10.6 | packaged by conda-forge | (main, Aug 22 2022, 20:41:54) [Clang 13.0.1 ]
Numpy 1.23.3
astropy 5.1
matplotlib 3.6.0
scipy 1.9.1
scikit-image 0.19.3
asdf 2.14.3
gwcs 0.18.2
regions 0.6
specutils 1.9.1
specreduce 1.3.1.dev3+g7be9a76
photutils 1.5.0
astroquery 0.4.6
pyyaml 6.0
asteval 0.9.28
idna 3.4
traitlets 5.4.0
bqplot 0.12.36
bqplot-image-gl 1.4.11
glue-core 1.6.0
glue-jupyter 0.14.2
glue-astronomy 0.5.1
echo 0.7
ipyvue 1.8.0
ipyvuetify 1.8.4
ipysplitpanes 0.2.0
ipygoldenlayout 0.4.0
Jinja2 3.1.2
voila 0.3.6
vispy 0.11.0
sidecar 0.5.1
Jdaviz 3.2.dev205+gd2282d27

[🐱](https://jira.stsci.edu/browse/JDAT-3120)

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to avoid redshift slider warning when setting `[cubeviz.]specviz.y_limits(...)`. The added test is confirmed to fail in current `main` without this patch.

I am not familiar with redshift slider logic, so please double check the code. Thanks!

Fixes #2029 

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/specviz/helper.py [hunk 1]
```diff
@@ -150,7 +150,11 @@ def y_limits(self, y_min=None, y_max=None):
         ref_index = self.app.get_viewer(
             self._default_spectrum_viewer_reference_name
         ).state.reference_data.label
-        flux_axis = self.get_spectra(ref_index).flux
+        if self.app.config == "cubeviz":
+            rs_opt = False
+        else:
+            rs_opt = True
```
Comment (kecnry) [inline jdaviz/configs/specviz/helper.py:156]: I would expect this to also happen in specviz and that we should always pass `False` here, no?

Comment (pllim) [reply jdaviz/configs/specviz/helper.py:156]: I dunno... What does True or False really mean?

Comment (pllim) [reply jdaviz/configs/specviz/helper.py:156]: The only reason is I don't know what is supposed to happen in other configs. So I am hoping this change will suppress warning but does not change any existing behavior. If you tell me setting to False always is safe, I will make that change. Please confirm. Thanks!

**Review by kecnry — APPROVED** [review]
LGTM - thanks!

**Review by rosteen — APPROVED** [review]
Simple enough, thanks.

**Comment by pllim** [comment]
Manual backport at #2063

**Pull request merged**